ee/clickhouse/models/property.py

```diff
@@ -22,19 +22,15 @@ def parse_prop_clauses(
             person_id_query, cohort_filter_params = format_filter_query(cohort)
             params = {**params, **cohort_filter_params}
             final += "AND distinct_id IN ({clause}) ".format(clause=person_id_query)
-
         elif prop.type == "person":
-
-            if json_extract:
-                filter_query, filter_params = prop_filter_json_extract(prop, idx, "person", "person_properties")
-                final += " {filter_query} AND team_id = %(team_id)s".format(filter_query=filter_query)
-                params.update(filter_params)
-            else:
-                filter_query, filter_params = person_prop_filter_kv(prop, idx)
-                final += " {filter_query}".format(filter_query=filter_query)
-                params.update(filter_params)
+            filter_query, filter_params = prop_filter_json_extract(prop, idx, "{}person".format(prepend))
+            final += " AND distinct_id IN ({filter_query})".format(
+                filter_query=GET_DISTINCT_IDS_BY_PROPERTY_SQL.format(
+                    filters=filter_query, negation="" if prop.operator and "not" in prop.operator else ""
```

Comment:
i'm might be wrong because the tests pass but something could be double negated here. Since prop_filter_json_extract should provide a query that includes 'NOT', there should be no need for negation here because before the conditions didn't include NOT directly and relied on this outside negation

Reply:
oh i see you removed the NOT, so this entire clause can be removed

Reply:
Great spot! Done and merged :)